pennylane/ops/functions/equal.py

```diff
@@ -25,12 +26,11 @@
 def equal(
     op1: Union[Operator, MeasurementProcess, ShadowMeasurementProcess],
     op2: Union[Operator, MeasurementProcess, ShadowMeasurementProcess],
-    check_interface=True,
-    check_trainability=True,
-    rtol=1e-5,
-    atol=1e-9,
+    **kwargs,
```

Comment:
Any particular reason to not explicitly write out the keyword arguments?

Reply:
This way the keyword arguments are only accepted in cases where they are actually applied. If they start all the way at the top and have to be passed through, then all of the sub-functions have to accept them, to prevent errors about receiving unexpected arguments. But setting them is only meaningful in the case of comparing two Operations.

So, this way, if you call `qml.equal(measurement_process1, measurement_process2, check_trainability=False)`, it will raise an error saying it doesn't accept that kwarg, whereas `qml.equal(operation1, operation2, check_trainability=False)` will work fine. If instead you move the kwargs up to here, it will just quietly accept `check_trainability=False` as an argument in all cases, even those that don't actually let you set it to False.

It also just looked tidier to only have them where they are used, in _equal_operation.

Reply:
I can see arguments for both ways. I think it comes down to some personal preference.

Even though pylint will complain, I'm fine with having unused keyword arguments if it means that the call signature remains consistent. 

The docstring (https://xanaduai-pennylane--3304.com.readthedocs.build/en/3304/code/api/pennylane.equal.html) and type hints (like with `? qml.equal`) will provide better information if the keyword arguments are written out in the call signature. I personally put more weight on having a clearly specified call signature we can see  in the documentation.

Reply:
I'll change it so they are specified at the top